tests/Graph/ConstantWeightShortestPathFinder.cpp

```diff
@@ -0,0 +1,421 @@
+      _toCondition->addMember(cond);
+    }
+
+    {
+      auto const* access =
+          ast->createNodeAttributeAccess(tmpId1, StaticStrings::FromString.c_str(),
+                                         StaticStrings::FromString.length());
+      auto const* cond =
+          ast->createNodeBinaryOperator(NODE_TYPE_OPERATOR_BINARY_EQ, access, tmpId2);
+      _fromCondition->addMember(cond);
+    }
+    spo = new ShortestPathOptions(query);
+    spo->setVariable(tmpVar);
+    spo->addLookupInfo(plan, "e", StaticStrings::FromString, _fromCondition->clone(ast));
+    spo->addReverseLookupInfo(plan, "e", StaticStrings::ToString, _toCondition->clone(ast));
+
+    spos.emplace_back(spo);
+    return spo;
+  }
+};
+
+}  // namespace
+
+namespace arangodb {
+namespace tests {
+namespace graph {
+
+TEST_CASE("ConstantWeightShortestPathFinder", "[graph]") {
+  Setup s;
+  UNUSED(s);
+  MockGraphDatabase gdb("testVocbase");
+
+  gdb.addVertexCollection("v", 100);
+  gdb.addEdgeCollection("e", "v",
+                        {{1, 2},   {2, 3},   {3, 4},   {5, 4},   {6, 5},
+                         {7, 6},   {8, 7},   {1, 10},  {10, 11}, {11, 12},
+                         {12, 4},  {12, 5},  {21, 22}, {22, 23}, {23, 24},
+                         {24, 25}, {21, 26}, {26, 27}, {27, 28}, {28, 25}});
+
+  auto query = gdb.getQuery("RETURN 1");
+  auto spo = gdb.getShortestPathOptions(query);
+
+  auto checkPath = [spo](ShortestPathResult result, std::vector<std::string> vertices,
+                         std::vector<std::pair<std::string, std::string>> edges) -> bool {
+    bool res = true;
+
+    if (result.length() != vertices.size()) return false;
+
+    for (size_t i = 0; i < result.length(); i++) {
+      auto vert = result.vertexToAqlValue(spo->cache(), i);
+      if (!vert.slice().get(StaticStrings::KeyString).isEqualString(vertices.at(i))) {
+        LOG_DEVEL << "expected vertex " << vertices.at(i) << " but found "
+                  << vert.slice().get(StaticStrings::KeyString).toString();
+        res = false;
+      }
+    }
+
+    // First edge is by convention null
+    CHECK((result.edgeToAqlValue(spo->cache(), 0).isNull(true)));
+    for (size_t i = 1; i < result.length(); i++) {
+      auto edge = result.edgeToAqlValue(spo->cache(), i);
+      if (!edge.slice().get(StaticStrings::FromString).isEqualString(edges.at(i).first) ||
+          !edge.slice().get(StaticStrings::ToString).isEqualString(edges.at(i).second)) {
+        LOG_DEVEL << "expected edge " << edges.at(i).first << " -> "
+                  << edges.at(i).second << " but found "
+                  << edge.slice().get(StaticStrings::FromString).toString() << " -> "
+                  << edge.slice().get(StaticStrings::ToString).toString();
+        res = false;
+      }
+    }
+    return res;
+  };
+
+  auto finder = new ConstantWeightShortestPathFinder(*spo);
+
+  // FIXME: passes even if vertex doesn't exist?
```

Comment:
FYI:

Yes, in ArangoDB vertices and edges are decoupled.
For the algorithms a non-existing vertex == null but the path is still valid.
So this is a no FIXME.
```suggestion

```